app/src/main/java/com/sdp13epfl2021/projmag/activities/SignInActivity.kt

```diff
@@ -91,6 +91,9 @@ class SignInActivity : AppCompatActivity() {
                             startActivity(intent)
                             finish()
                         } else if(isNew == false) {
+
+                            val settings = getSharedPreferences(UserTypeChoice.savedTypeChoice, Context.MODE_PRIVATE)
+                            UserTypeChoice.isProfessor = settings.getBoolean(UserTypeChoice.isUserProf,true)
```

Comment:
The isProfessor should already be initialized by the code in MainActivity.
Maybe this code can be removed